onnxruntime/core/providers/cpu/controlflow/scan_8.cc

```diff
@@ -396,13 +396,13 @@ Status Scan8Impl::Execute(const FeedsFetchesManager& ffm) {
 
     // Setup input OrtValue streams
     std::vector<OrtValueTensorSlicer<const OrtValue>::Iterator> scan_input_stream_iterators;
-    scan_input_stream_iterators.reserve(info_.num_variadic_inputs - info_.num_loop_state_variables);
+    scan_input_stream_iterators.reserve(static_cast<size_t>(info_.num_variadic_inputs) - info_.num_loop_state_variables);
 
     for (int i = info_.num_loop_state_variables, end = info_.num_variadic_inputs; i < end; ++i) {
       const auto& ort_value = GetSubgraphInputMLValue(context_, i);
 
       // forward
-      if (directions_[i - info_.num_loop_state_variables] == static_cast<int64_t>(ScanDirection::kForward)) {
+      if (directions_[static_cast<ptrdiff_t>(i) - info_.num_loop_state_variables] == static_cast<int64_t>(ScanDirection::kForward)) {
```

Comment:
It is safe to convert int to ptrdiff_t, because ptrdiff_t is always bigger. But when we convert it to size_t, we must be sure the integer is non-negative. In the case in loop.cc, because i starts from 0 and it always increases, so it is obvious. But I don't know if num_loop_carried_vars is signed or not, so I can't change the type of i. 

In this case, I'm not that sure.  Also, even if I know i is unsigned, I need to be sure it is always greater than info_.num_loop_state_variables too. Otherwise if I changed it to unsigned, the result would be a mess.